app/design/frontend/base/default/template/razorpay/extensions/firecheckout/payments.phtml

```diff
@@ -0,0 +1,66 @@
+<?php $helper = Mage::helper('razorpay_payments'); ?>
+<?php if ($helper->isRazorpayEnabled()) : ?>
+    <?php $code = Razorpay_Payments_Model_Paymentmethod::METHOD_CODE ; ?>
+    <script type="text/javascript">
+        var formId = 'firecheckout-form';
+        var paymentIdField = 'razorpay_payment_id';
+        var nativeButton = $$('#review-buttons-container .btn-checkout').first();
+
+        // Change native extension place order button onclick
+        nativeButton.setAttribute('onclick', 'processRazorpayOrder()');
+
+        // Click extension "place order" button
+        var processNativeExtensionOrder = function(){
+            nativeButton.show();
+
+            checkout.save();
+        };
+
+        var showErrorMessage = function(message){
+            alert(message);
+
+            nativeButton.show();
+        };
+
+        var processRazorpayOrder = function() {
+            if (payment.currentMethod == '<?php echo $code ?>') {
+                var $paymentId = $(paymentIdField);
+
+                var captureOrder = $paymentId && ($paymentId.value != '');
```

Comment:
Since you are using jQuery, you can just do `$(paymentIdField).val() !== ''`. I'd prefer triple equality here. `.val() ` can return undefined as well, if the field is not present in the HTML. 

On second look, both `$paymentId` and `captureOrder` are unused.